src/http/h1/decode.rs

```diff
@@ -290,4 +292,40 @@ mod tests {
         assert_eq!(e.kind(), io::ErrorKind::Other);
         assert_eq!(e.description(), "early eof");
     }
+
+    fn read_chunked_async(content: &[u8], block_at: usize, read_buffer_size: usize) -> String {
+        let content_len = content.len();
+        let mock_buf = io::Cursor::new(content.clone());
+        let mut ins = Async::new(mock_buf, block_at);
+        let mut decoder = Decoder::chunked();
```

Comment:
Can probably have 'decode' as a argument here, and make this just a `read_async` function to test the other decoders as well perhaps.